model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -589,143 +609,122 @@ def get_config_options_by_operators_set(self,
-            return None
-
-        return opset_list[0]  # There's one opset with that name
+        if len(opset_list) > 1:
+            Logger.critical(f"Found more than one OperatorsSet in TargetPlatformModel with the name {opset_name}.")
+        return opset_list[0] if opset_list else None
 
-    def append_component(self,
-                         tp_model_component: TargetPlatformModelComponent):
+    def append_component(self, tp_model_component: TargetPlatformModelComponent) -> None:
         """
-        Attach a TargetPlatformModel component to the model. Components can be for example:
-        Fusing, OperatorsSet, etc.
+        Attach a TargetPlatformModel component to the model (like Fusing or OperatorsSet).
 
         Args:
-            tp_model_component: Component to attach to the model.
+            tp_model_component (TargetPlatformModelComponent): Component to attach to the model.
 
+        Raises:
+            Logger critical if the component is not an instance of Fusing or OperatorsSetBase.
         """
         if isinstance(tp_model_component, Fusing):
             self.fusing_patterns.append(tp_model_component)
         elif isinstance(tp_model_component, OperatorsSetBase):
             self.operator_set.append(tp_model_component)
-        else:  # pragma: no cover
-            Logger.critical(f'Attempted to append an unrecognized TargetPlatformModelComponent of type: {type(tp_model_component)}.')
+        else:
+            Logger.critical(
+                f"Attempted to append an unrecognized TargetPlatformModelComponent of type: {type(tp_model_component)}.") # pragma: no cover
 
-    def __enter__(self):
+    def get_info(self) -> Dict[str, Any]:
         """
-        Start defining the TargetPlatformModel using 'with'.
-
-        Returns: Initialized TargetPlatformModel object.
+        Get a dictionary summarizing the TargetPlatformModel properties.
 
+        Returns:
+            Dict[str, Any]: Summary of the TargetPlatformModel properties.
         """
-        _current_tp_model.set(self)
-        return self
-
-    def __exit__(self, exc_type, exc_value, tb):
-        """
-        Finish defining the TargetPlatformModel at the end of the 'with' clause.
-        Returns the final and immutable TargetPlatformModel instance.
-        """
+        return {
+            "Model name": self.name,
+            "Default quantization config": self.get_default_op_quantization_config().get_info(),
+            "Operators sets": [o.get_info() for o in self.operator_set],
+            "Fusing patterns": [f.get_info() for f in self.fusing_patterns],
+        }
 
-        if exc_value is not None:
-            print(exc_value, exc_value.args)
-            raise exc_value
-        self.__validate_model()  # Assert that model is valid.
-        _current_tp_model.reset()
-        self.initialized_done()  # Make model immutable.
-        return self
-
-    def __validate_model(self):
+    def __validate_model(self) -> None:
```

Comment:
Why do you need both post_init and validate?

Reply:
I agree. Currently, this is part of the context manager. It will be removed in the next PR when the context manager mechanism is removed.